matchbox_socket/src/webrtc_socket/error.rs

```diff
@@ -45,15 +47,25 @@ pub enum SignalingError {
 }
 
 /// An error that can occur with WebRTC messaging.
-#[cfg(not(target_arch = "wasm32"))]
 #[derive(Debug, thiserror::Error)]
-#[error("failed to send message to peer")]
-pub(crate) struct MessagingError(#[from] futures_channel::mpsc::TrySendError<crate::Packet>);
+pub enum MessageSendError {
+    #[error("Failed to send id to peer {0}")]
+    PeerId(#[from] crossbeam_channel::TrySendError<PeerId>),
 
-#[cfg(target_arch = "wasm32")]
-#[derive(Debug, thiserror::Error)]
-#[error("failed to send message to peer")]
-pub(crate) struct MessagingError(#[from] JsError);
+    #[error("Failed to send report peer state ({}) to peer {}", 0.1, 0.0)]
+    PeerState(#[from] TrySendError<(PeerId, PeerState)>),
+
+    #[error("Failed to send request to peer")]
+    Request(#[from] TrySendError<PeerRequest>),
+
+    #[cfg(not(target_arch = "wasm32"))]
+    #[error("Failed to send message to peer")]
+    Packet(#[from] TrySendError<crate::Packet>),
```

Comment:
Are most these just the result of the socket being dropped before the future (some `UnboundedSender::try_send` fails)?

Do the user really care (and want to handle) all these different types of errors separately?

Maybe it makes more sense to hide this from public API and just have a socket dropped or similar error? Wondering if we're kind of leaking irrelevant internals here...

Reply:
They are all the result of a try_send failing, yes. I agree with making one error rather than fragmenting.